Bottom: 8faf883fe5de08676d01f7cd7448f3c5a0d673c5
Top:    6e79acc95e5cd324e988850c7bb7a5ba4b71e9ee
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-07 14:20:46 +0100

Generate pcb-menu.res and gpcb-menu.res from ".in" files

No functional changes here, just a preliminary cleaning before
adding the topological autorouter to the menus.

Based on a patch by Stanislav Brabec <sbrabec@suse.cz>

Affects-bug: lp-812429


---

diff --git a/src/Makefile.am b/src/Makefile.am
index 7efb2d7..f57483c 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -18,7 +18,7 @@ EXTRA_LIBRARIES = \
 	liblpr.a libgerber.a libbom.a libpng.a libps.a libnelma.a \
 	libgcode.a
 
-pcblib_DATA=	\
+pcblib_DATA= \
 	default_font \
 	gpcb-menu.res \
 	pcb-menu.res
@@ -230,8 +230,8 @@ EXTRA_DIST= \
 	$(srcdir)/hid/png/hid.conf \
 	$(srcdir)/hid/nelma/hid.conf \
 	$(srcdir)/hid/ps/hid.conf \
-	gpcb-menu.res \
-	pcb-menu.res \
+	gpcb-menu.res.in \
+	pcb-menu.res.in \
 	pcbtest.sh.in \
 	dbus.xml
 
@@ -239,12 +239,20 @@ AM_YFLAGS=	-d
 
 all-local: pcbtest.sh
 
+pcb-menu.res : pcb-menu.res.in
+	echo '/* AUTOMATICALLY GENERATED FROM pcb-menu.res.in DO NOT EDIT */' > $@
+	cat ${srcdir}/pcb-menu.res.in >> $@
+
 pcb-menu.h : pcb-menu.res
 	echo '/* AUTOMATICALLY GENERATED FROM pcb-menu.res DO NOT EDIT */' > $@
 	echo 'const char *pcb_menu_default[] = {' >> $@
 	sed 's/\\/\\\\/g; s/"/\\"/g; s/^/"/; s/$$/",/' < ${srcdir}/pcb-menu.res >> $@
 	echo '0};' >> $@
 
+gpcb-menu.res : gpcb-menu.res.in
+	echo '/* AUTOMATICALLY GENERATED FROM gpcb-menu.res.in DO NOT EDIT */' > $@
+	cat ${srcdir}/gpcb-menu.res.in >> $@
+
 gpcb-menu.h : gpcb-menu.res
 	echo '/* AUTOMATICALLY GENERATED FROM gpcb-menu.res DO NOT EDIT */' > $@
 	echo 'const char *gpcb_menu_default[] = {' >> $@
@@ -449,7 +457,9 @@ DISTCLEANFILES= pcbtest.sh gpcb-menu.h pcb-menu.h \
 	hid/nelma/nelma_lists.h \
 	hid/ps/ps_lists.h \
 	core_lists.h \
-	dbus-introspect.h
+	dbus-introspect.h \
+	gpcb-menu.res \
+	pcb-menu.res
 
 # create wrapper script that lets you test pcb prior to installation
 pcbtest.sh:	$(srcdir)/pcbtest.sh.in Makefile
diff --git a/src/gpcb-menu.res b/src/gpcb-menu.res.in
similarity index 100%
rename from src/gpcb-menu.res
rename to src/gpcb-menu.res.in
diff --git a/src/pcb-menu.res b/src/pcb-menu.res.in
similarity index 100%
rename from src/pcb-menu.res
rename to src/pcb-menu.res.in
